config/src/test/kotlin/org/springframework/security/config/web/servlet/SessionManagementDslTests.kt

```diff
@@ -29,31 +34,76 @@ import org.springframework.security.config.annotation.web.configuration.WebSecur
 import org.springframework.security.config.http.SessionCreationPolicy
 import org.springframework.security.config.test.SpringTestRule
 import org.springframework.security.core.Authentication
+import org.springframework.security.core.GrantedAuthority
+import org.springframework.security.core.context.SecurityContext
 import org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.authentication
 import org.springframework.security.web.authentication.SimpleUrlAuthenticationFailureHandler
 import org.springframework.security.web.authentication.session.SessionAuthenticationException
 import org.springframework.security.web.authentication.session.SessionAuthenticationStrategy
+import org.springframework.security.web.context.HttpSessionSecurityContextRepository
+import org.springframework.security.web.csrf.CsrfToken
+import org.springframework.security.web.csrf.HttpSessionCsrfTokenRepository
+import org.springframework.security.web.savedrequest.HttpSessionRequestCache
+import org.springframework.security.web.savedrequest.SavedRequest
 import org.springframework.security.web.session.SimpleRedirectInvalidSessionStrategy
 import org.springframework.test.web.servlet.MockMvc
 import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers.redirectedUrl
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
-import javax.servlet.http.HttpServletRequest
-import javax.servlet.http.HttpServletResponse
+import org.springframework.web.servlet.support.SessionFlashMapManager
 
 /**
  * Tests for [SessionManagementDsl]
  *
  * @author Eleftheria Stein
  */
 class SessionManagementDslTests {
+
+    companion object {
+        private const val SAVED_REQUEST_ATTR_NAME = "SPRING_SECURITY_SAVED_REQUEST"
+
+        private val DEFAULT_CSRF_TOKEN_ATTR_NAME = "${HttpSessionCsrfTokenRepository::class.java.name}.CSRF_TOKEN"
+        private val FLASH_MAPS_SESSION_ATTR_NAME = "${SessionFlashMapManager::class.java.name}.FLASH_MAPS"
+    }
+
     @Rule
     @JvmField
     val spring = SpringTestRule()
 
     @Autowired
     lateinit var mockMvc: MockMvc
 
+    private val mockedCsrfToken: CsrfToken = mockk(relaxed = true)
+    private val mockedSavedRequest: SavedRequest = mockk(relaxed = true)
+    private val mockedSecurityContext: SecurityContext = mockk(relaxed = true)
+    private val mockedAuthentication: Authentication = mockk(relaxed = true)
+    private val session: MockHttpSession = mockk()
+
+    @Before
+    fun setUp() {
+        every { session.changeSessionId() } throws SessionAuthenticationException("any SessionAuthenticationException")
```

Comment:
Let's keep the mocks in each test in order to avoid test pollution.

Reply:
I'll will change it in the next commit.